src/types/redis_zset.h

```diff
@@ -35,46 +36,34 @@ const double kMinScore = (std::numeric_limits<float>::is_iec559 ? -std::numeric_
 const double kMaxScore = (std::numeric_limits<float>::is_iec559 ? std::numeric_limits<double>::infinity()
                                                                 : std::numeric_limits<double>::max());
 
-typedef struct ZRangeSpec {
-  double min, max;
-  bool minex, maxex; /* are min or max exclusive */
-  int offset, count;
-  bool removed, reversed;
-  ZRangeSpec() {
-    min = kMinScore;
-    max = kMaxScore;
-    minex = maxex = false;
-    offset = -1;
-    count = -1;
-    removed = reversed = false;
-  }
-} ZRangeSpec;
-
-typedef struct ZRangeLexSpec {
+struct ZrangeCommon {
```

Comment:
@tanruixiang I prefer the old way to separate `ZRangeSpec` and `ZRangeLexSpec`, it's more clear when parsing even with a bit of duplicate code and we also don't need to use the ptr in the Parse function.

Reply:
We have three modes, and the possibility of increasing the options int future. If we don't use inheritance, the code is simply copied and pasted in three copies. Are there any other pitfalls in using pointers? @PragmaTwice